src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -329,7 +329,19 @@ private static string GetVsRootFromMSBuildAssembly(string msBuildAssembly)
 
         private static string GetMSBuildExeFromVsRoot(string visualStudioRoot)
         {
-            return FileUtilities.CombinePaths(visualStudioRoot, "MSBuild", CurrentToolsVersion, "Bin", "MSBuild.exe");
+            return FileUtilities.CombinePaths(
+                visualStudioRoot,
+                "MSBuild",
+                CurrentToolsVersion,
+                "Bin",
+#if NET35
+                IntPtr.Size == 8
+#else
+                System.Runtime.InteropServices.Marshal.SizeOf<IntPtr>() == 8
+#endif
+                  ? "amd64"
+                  : string.Empty,
```

Comment:
I imagine this will create a double-slash, but I don't think there's a way around that that improves performance.

Reply:
Probably doing two if-then-else Combine calls depending on bitness would be faster than the unfathomable machinery triggered in .net and windows by the two `\\`

Reply:
I was curious about this as I implemented it. It actually does The Right Thing:

```sh-session
$ type .\Program.cs
using System;

namespace path_thing
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine(System.IO.Path.Combine("a", "b", string.Empty, "c"));
        }
    }
}

/play/path-thing via •NET v3.1.200-preview-015002
$ dotnet run
a\b\c
```

Reply:
Curious, in CSI I get:

```c#
> System.IO.Path.Combine("a", string.Empty, "b")
"a\\b"
```

Reply:
Ah, CSI pretty prints the string and escapes the slash, which I judge as a very dubious choice

```c#
> var a = System.IO.Path.Combine("a", "", "b");
> foreach (char i in a) System.Console.WriteLine(i);
a
\
b
```